receiver.go

```diff
@@ -249,14 +249,21 @@ func (r *Receiver) sendDisposition(ctx context.Context, first uint32, last *uint
 		State:   state,
 	}
 
+	sent := make(chan error, 1)
 	select {
-	case r.txDisposition <- fr:
-		debug.Log(2, "TX (Receiver %p): mux txDisposition %s", r, fr)
-		return nil
 	case <-r.l.done:
 		return r.l.doneErr
 	case <-ctx.Done():
```

Comment:
As it turns out, this part of the change is responsible for the bug we've been seeing in stress.  Previously, only waiting for the disposition ack was cancelable.  Now, the cancel can happen before or after the disposition frame has been sent, leaving callers in an unknown state.  I think this part of the change should be reverted so that only waiting for the ack is cancelable.

Reply:
This has been removed.

Reply:
You tricked me again, Hendrix. Probaby not for the last time.